### PR TITLE
Button: restore "inactive" prop; typescript-ify

### PR DESCRIPTION
**Pull request by adamhooper**
Relates to QZ-151 in that I tried calling `<Button>` from typescript and this error appeared.

It looks like "inactive" might have been removed in error. `qz-react` uses it in seven different calls.

### CODE REVIEW

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -1,155 +1,151 @@
...
 	 * https://amp.dev/documentation/examples/components/amp-bind/
 	 */
-	on: PropTypes.string,
+	on?: string,
 	/**
 	 * Function to run when the button is engaged. Forwarded to the button
 	 * element.
 	 */
-	onClick: PropTypes.func.isRequired,
+	onClick?: React.UIEventHandler<HTMLButtonElement>,
```
Comment (Bhammy) [inline src/components/Button/Button.tsx:93]: Hmm - why make onClick optional? It seems like a useful pointer to say that maybe another kind of element is required

**Comment by adamhooper** [comment]
An HTML default button type is "submit", which means, "submit the <form>".
qz-react had a couple of forms that use this default (and omit onClick).

... Ironically, the QZ-151 bug that started my journey involves this
default behavior happening on top of the onClick :)

File: src/components/Button/Button.tsx (hunk 1 above)
Comment (adamhooper) [reply src/components/Button/Button.tsx]: A `<button type="submit">` doesn't need (and shouldn't have) an `onClick`.

I think this nit should be obvious when _calling_ the component. But when I converted to Typescript, I erroneously set this to "required" and then had to backtrack when the callers threw type-safety errors. So I've added an inline comment to save the next developer the same embarrassment.

Comment (Bhammy) [reply src/components/Button/Button.tsx]: 🤦 of course! Good points.

**Review by Bhammy — APPROVED** [review]

**Pull request merged**